File: 프로그래머스/unrated/134240. 푸드 파이트 대회/푸드 파이트 대회.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

string solution(vector<int> food)
{
    string answer = "";
    string temp = "";
    int len = 0;
    for (int i = 1; i < food.size(); i++)
    {
        len = food[i] / 2;
        if (len >= 1)
        {
            int z = 0;
            while (z < len)
            {
                temp += to_string(i);
                z++;
            }
        }
    }
    answer += temp;
    reverse(temp.begin(), temp.end());
    answer += '0';
    answer += temp;

    return answer;
}

File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.java
class Solution {

    public String solution(int[] food) {
        StringBuilder sb = new StringBuilder();

        for (int i = 1, idx = 1; i < food.length; i++, idx++) {
            for (int j = 0; j < food[i] / 2; j++) {
                sb.append(idx);
            }
        }

        return sb.toString() + sb.append("0").reverse();
    }
}